src/utils/helper-functions/versions.ts

```diff
@@ -0,0 +1,23 @@
+import * as semver from 'semver'
+
+let ngrxEffectsVersion: string | undefined
+function getNgrxEffectsVersion(): string {
+  if (ngrxEffectsVersion) return ngrxEffectsVersion
+
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const effects = require('@ngrx/effects/schematics-core') as {
+      platformVersion: string
+    }
+    // version in file is ^x.x.x, but this isn't valid for semver
+    ngrxEffectsVersion = effects.platformVersion?.substr(1)
+  } catch (err) {
+    ngrxEffectsVersion = '0.0.0'
+  }
+
+  return ngrxEffectsVersion ?? '0.0.0'
+}
+
+export function ngrxEffectsVersionSatisfies(version: string) {
+  return semver.satisfies(getNgrxEffectsVersion(), version)
+}
```

Comment:
Instead of _memoize_ only the `ngrxEffectsVersion`, I'd _memoize_ this whole function (and remove the `ngrxEffectsVersion` of the top):

```suggestion
let effectsVersionSatisfies: boolean | undefined
export function ngrxEffectsVersionSatisfies(version: string): boolean {
  return (
    effectsVersionSatisfies ??
    (effectsVersionSatisfies = semver.satisfies(
      getNgrxEffectsVersion(),
      version,
    ))
  )
}
```

Reply:
I think that this won't work if we have different rules working on different verion requirements?

Reply:
You got a point 😅.

What about this update to `ngrxEffectsVersionSatisfies`?

```ts
const effectsVersionSatisfies: Record<string, boolean> = {}
export function ngrxEffectsVersionSatisfies(version: string) {
  return (
    effectsVersionSatisfies[version] ??
    (effectsVersionSatisfies[version] = semver.satisfies(getVersion(), version))
  )
}

```

Reply:
Eh, I added it but I had some struggles to make it work during tests.
I thought each test file would have been isolated but it seems like that's not the case.
Thus, it required a bit more setup - let me know what you think